addons/survey/controllers/main.py

```diff
@@ -186,7 +180,7 @@ def _prepare_retry_additional_values(self, answer):
         }
 
     def _prepare_survey_finished_values(self, survey, answer, token=False):
-        values = {'survey': survey, 'answer': answer}
+        values = {'survey': survey, 'answer': answer, 'finished': True}
```

Comment:
I already started to remove 'finished' param in templates but I guess I forgot to do it in controllers. This could be removed too to use answer_sudo instead in _prepare_question_html